sycl/test/fpga_tests/compute_units.cpp

```diff
@@ -0,0 +1,81 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+//==------------ compute_units.cpp - SYCL FPGA compute units test ----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl/INTEL/fpga_extensions.hpp>
+
+using namespace sycl;
+
+constexpr float kTestData = 555;
+
+constexpr size_t kComputeUnits = 5;
+using Pipes = INTEL::PipeArray<class MyPipe, float, 4, kComputeUnits + 1>;
+
+class source_kernel;
+class sink_kernel;
+
+template <std::size_t ID> class chain_kernel;
+
+// Write the first piece of data to the pipeline
+void SourceKernel(queue &q, float data) {
+
+  q.submit([&](handler &h) {
+    h.single_task<source_kernel>([=]() { Pipes::PipeAt<0>::write(data); });
```

Comment:
```suggestion
    h.single_task<source_kernel>([=] { Pipes::PipeAt<0>::write(data); });
```
Otherwise, I was thinking to this alternative API:
```c++
    h.single_task<source_kernel>([=] { Pipes::write<0>(data); });
```